src/metabase/sync/sync_metadata/sync_table_privileges.clj

```diff
@@ -0,0 +1,34 @@
+(ns metabase.sync.sync-metadata.sync-table-privileges
+  (:require
+   [clojure.java.jdbc :as jdbc]
+   [honey.sql :as sql]
+   [metabase.driver :as driver]
+   [metabase.driver.util :as driver.u]
+   [metabase.models.interface :as mi]
+   [metabase.util.malli :as mu]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(mu/defn sync-table-privileges!
+  "Sync the `table_privileges` table with the privileges in the database.
+
+   This is a cache of the data returned from `driver/table-privileges`, but it's stored in the database for performance."
+  [database :- (mi/InstanceOf :model/Database)]
+  (let [driver (driver.u/database->driver database)]
+    (when (contains? (methods driver/current-user-table-privileges) driver)
```

Comment:
if we add `:table-privileges` to `driver/drvier-feaures` then we can do `driver/supports :table-privileges` here

Reply:
This will be replaced by `driver/supports :actions`. I've left a comment here cf8a953e076ca194f42275261b80aa4e9652a35a